### PR TITLE
feat: added cache to main function

### PR DESCRIPTION
**Pull request by holytimy4**

### CODE REVIEW

File: config_parser.js [hunk 1]
```diff
@@ -4,32 +4,55 @@ const path = require('path');
 
 const ajv = new Ajv({ allErrors: true, useDefaults: true });
 
-function loadConfig(configPath, schemaPath, defaultValues = {}, preprocess = null) {
+let configCache = {};
+
+function loadConfig(configPath, schemaPath, defaultValues = {}, preprocess = null, options = { cache: true, log: true }) {
```
Comment (VadimOpt) [inline config_parser.js:9]: Good job! Added caching and logging options to loadConfig function. Very smart

**Review by VadimOpt — COMMENTED** [review]
Described below

File: config_parser.js [hunk 2]
```diff
@@ -4,32 +4,55 @@ const path = require('path');
...
         if (preprocess) {
-            console.log('Preprocessing configuration...');
+            log('Preprocessing configuration...');
             preprocess(finalConfig);
         }
 
-        console.log('Config validated and loaded successfully.');
+        log('Config validated and loaded successfully.');
+
+        if (options.cache) {
```
Comment (gfdsokfsdn3) [inline config_parser.js:52]: Well done. Cache comparison seems smart.

**Review by gfdsokfsdn3 — COMMENTED** [review]
Add a comment

**Pull request merged**